x/incentives/keeper/gauge.go

```diff
@@ -287,12 +286,19 @@ func (k Keeper) GetEpochInfo(ctx sdk.Context) epochtypes.EpochInfo {
 // The fee is sent to the community pool.
 // Returns nil on success, error otherwise.
 func (k Keeper) chargeFeeIfSufficientFeeDenomBalance(ctx sdk.Context, address sdk.AccAddress, fee sdk.Int, gaugeCoins sdk.Coins) (err error) {
-	totalCost := gaugeCoins.AmountOf(appparams.BaseCoinUnit).Add(fee)
-	accountBalance := k.bk.GetBalance(ctx, address, appparams.BaseCoinUnit).Amount
+	feeDenom, err := k.tk.GetBaseDenom(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
This is technically a state break but the likelihood of this non-deterministically happening is low so I think we can still backport this

Reply:
Actually, since this is on the message path, it is possible to run out of gas. So I think we should avoid backporting